packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -605,6 +605,60 @@ export default class RestApi {
       res.redirect(308, req.baseUrl + '/backup/jobs/vm/' + req.params.id)
     })
 
+    api.get(
+      '/dashboard',
+      wrap(async (req, res) => {
+        const dashboard = {}
+
+        const poolIds = new Set()
+        const hosts = []
+
+        for (const obj of app.objects.values()) {
+          if (obj.type === 'host') {
+            hosts.push(obj)
+            poolIds.add(obj.$pool)
+          }
+        }
+
+        dashboard.nPools = poolIds.size
+        dashboard.nHosts = hosts.length
+
+        let listMissingPatchesAuthorized = true
+        try {
+          await app.checkFeatureAuthorization('LIST_MISSING_PATCHES')
+        } catch (error) {
+          listMissingPatchesAuthorized = false
+        }
+
+        if (listMissingPatchesAuthorized) {
+          const poolsWithMissingPatches = new Set()
+          let nHostsWithMissingPatches = 0
+
+          await asyncEach(hosts, async host => {
+            const xapi = app.getXapi(host)
+            try {
+              const patches = await xapi.listMissingPatches(host)
+              if (patches.length > 0) {
+                nHostsWithMissingPatches++
+                poolsWithMissingPatches.add(host.$pool)
+              }
+            } catch (error) {
+              console.error(error)
+            }
+          })
+
+          const missingPatches = {
+            nHostsWithMissingPatches,
+            nPoolsWithMissingPatches: poolsWithMissingPatches.size,
+          }
+
+          dashboard.missingPatches = missingPatches
```

Comment:
I believe we to implement a `hasFeatureAuthorization()` which returns a boolean instead of throwing, it will make this code much clearer.